File: benchmarks/primitives/peers/int_bitops.zig
const std = @import("std");
fn now() f64 {
    var ts: std.c.timespec = undefined;
    _ = std.c.clock_gettime(std.c.CLOCK.MONOTONIC, &ts);
    return @as(f64, @floatFromInt(ts.sec)) + @as(f64, @floatFromInt(ts.nsec)) / 1e9;
}
pub fn main() void {
    const n: i64 = 300000000; var s: i64 = 1;
    const t0 = now(); var i: i64 = 0;
    while (i < n) : (i += 1) { s = (s << @as(u6, 13)) ^ (s >> @as(u6, 7)) ^ i; }
    const el = now() - t0;
    var buf: [256]u8 = undefined;
    const out = std.fmt.bufPrint(&buf, "{d}\nops: {d}\nelapsed: {d:.6}s\n", .{ s, n, el }) catch return;
    _ = std.c.write(1, out.ptr, out.len);
}

File: benchmarks/primitives/peers/int_bitops.rs
use std::time::Instant;
fn main(){ let n:i64=300000000; let mut s:i64=1; let t0=Instant::now();
 for i in 0..n { s=(s<<13)^(s>>7)^i; }
 let el=t0.elapsed().as_secs_f64();
 println!("{}\nops: {}\nelapsed: {:.6}s",s,n,el); }
